src/net/p2p.rs

```diff
@@ -4276,7 +4276,15 @@ impl PeerNetwork {
         }
 
         if let Err(e) = PeerNetwork::setup_unconfirmed_state(chainstate, sortdb) {
-            warn!("Failed to instantiate unconfirmed state: {:?}", &e);
+            if let net_error::ChainstateError(ref err_msg) = e {
+                if err_msg == "Stacks chainstate error: NoSuchBlockError" {
```

Comment:
This makes me wonder if there should just be a dedicated error type for this particular condition.

Reply:
Perhaps, but I'd leave refactoring this error variant to a different PR (this would be less hacky if the `net_error::ChainstateError` variant took a `ChainstateError` enum rather than a String).